### PR TITLE
spec(abci): slight improvements to ABCI spec (@angrav's feedback from Anoma)

### PR DESCRIPTION
**Pull request by sergio-mena**
---

#### PR checklist

- [ ] ~~Tests written/updated~~
- [ ] ~~Changelog entry added in `.changelog` (we use [unclog](https://github.com/informalsystems/unclog) to manage our changelog)~~
- [x] Updated relevant documentation (`docs/` or `spec/`) and code comments
- [x] Title follows the [Conventional Commits](https://www.conventionalcommits.org/en/v1.0.0/) spec

### CODE REVIEW

**Review by melekes — APPROVED** [review]
👍

File: spec/abci/abci++_comet_expected_behavior.md [hunk 1]
```diff
@@ -157,14 +157,17 @@ did not store any state CometBFT calls `InitChain`. After this, CometBFT enters
   `FinalizeBlock`, followed by a call to `Commit`. In each round, the sequence of method calls
   depends on whether the local process is the proposer or not. Note that, if a height contains zero
   rounds, this means the process is replaying an already decided value (catch-up mode).
+  When calling `FinalizeBLock` with a block, the consensus algorithm run by CometBFT guarantees
```
Comment (andynog) [inline spec/abci/abci++_comet_expected_behavior.md:160]:
```suggestion
  When calling `FinalizeBlock` with a block, the consensus algorithm run by CometBFT guarantees
```

File: spec/abci/abci++_methods.md [hunk 2]
```diff
@@ -661,6 +661,8 @@ without calling `VerifyVoteExtension` to verify it.
       making the Application's state evolve in the context of state machine replication.
     * Currently, CometBFT will fill up all fields in `FinalizeBlockRequest`, even if they were
       already passed on to the Application via `PrepareProposalRequest` or `ProcessProposalRequest`.
+    * When calling `FinalizeBLock` with a block, the consensus algorithm run by CometBFT guarantees
```
Comment (andynog) [inline spec/abci/abci++_methods.md:664]:
```suggestion
    * When calling `FinalizeBlock` with a block, the consensus algorithm run by CometBFT guarantees
```

File: spec/abci/abci++_basic_concepts.md [hunk 3]
```diff
@@ -123,6 +123,8 @@ call sequences of these methods.
   update its state accordingly. Cryptographic commitments to the block and transaction results,
   returned via the corresponding parameters in `FinalizeBlockResponse`, are included in the header
   of the next block. CometBFT calls it when a new block is decided.
+  When calling `FinalizeBLock` with a block, the consensus algorithm run by CometBFT guarantees
```
Comment (andynog) [inline spec/abci/abci++_basic_concepts.md:126]:
```suggestion
  When calling `FinalizeBlock` with a block, the consensus algorithm run by CometBFT guarantees
```

**Review by andynog — APPROVED** [review]
lgtm, just minor typo fixes suggestions

**Pull request merged**